### PR TITLE
`ultralytics 8.3.123` Rockchip RKNN export INT8 quantization support

### PR DESCRIPTION
**Pull request by oDestroyeRo**
#20144  Add INT8 quantization support for RKNN model export

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
Added INT8 quantization support for RKNN model exports, improving performance and flexibility for Rockchip devices. 🚀

### 📊 Key Changes
- Introduced a new `int8` export argument for RKNN, allowing users to export models in INT8 quantized format.
- Updated documentation and export tables to reflect the new `int8` option for RKNN exports.
- Modified the export process so exported RKNN model filenames now indicate whether they are INT8 or FP16.
- Bumped the Ultralytics package version to 8.3.123.

### 🎯 Purpose & Impact
- **Better Performance:** INT8 quantization can significantly reduce model size and speed up inference on compatible Rockchip hardware.
- **More Flexibility:** Users can now choose between FP16 and INT8 exports for RKNN, tailoring models to their deployment needs.
- **Clearer Exports:** Updated filenames make it easy to identify the model type (INT8 or FP16), reducing confusion during deployment.
- **Improved Documentation:** Clearer guidance and updated tables help users understand and utilize the new feature quickly.

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @oDestroyeRo, thank you for submitting an `ultralytics/ultralytics` 🚀 PR! To ensure a seamless integration of your work, please review the following checklist:

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_  — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). Don’t hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

_(This is an automated response. An Ultralytics engineer will assist you soon.)_

**Comment by oDestroyeRo** [comment]
I have read the CLA Document and I sign the CLA

**Comment by UltralyticsAssistant** [comment]
Exciting news—this PR is now merged! 🚀 Thank you, @oDestroyeRo and @glenn-jocher, for advancing INT8 quantization support for RKNN exports and enriching the Ultralytics ecosystem. As Thomas Edison once said, "There's a way to do it better—find it." Your efforts embody this spirit, empowering users with greater performance and flexibility on Rockchip devices. We truly appreciate your dedication and innovation—together, we're making Ultralytics HUB and YOLO11 even more impactful!

**Comment by yg-dickson** [comment]
The command I used to export rknn int8 is:
yolo export model=best.pt format=rknn int8=true name=rk3588

E build: If set do_quantization = True or auto_hybrid = True, dataset file need be provided. But it`s value is None!
E export_rknn: RKNN model does not exist, please load & build model first!

**Comment by glenn-jocher** [comment]
Hi @yg-dickson, the error you're encountering occurs because INT8 quantization requires a dataset for calibration. You need to add a `data` parameter to your export command:

```
yolo export model=best.pt format=rknn int8=True name=rk3588 data=path/to/dataset.yaml
```

The dataset is needed for calibration to determine optimal quantization scales. If you don't have a specific dataset, you can use one of the default datasets like `coco8.yaml` which will be downloaded automatically:

```
yolo export model=best.pt format=rknn int8=True name=rk3588 data=coco8.yaml
```

This should resolve the error you're experiencing with the RKNN export process.

**Comment by yg-dickson** [comment]
I use yolo export model=best.pt format=rknn int8=True name=rk3588 data=coco8.yaml but still get an error

E build: If set do_quantization = True or auto_hybrid = True, dataset file need be provided. But it`s value is None!
E export_rknn: RKNN model does not exist, please load & build model first!

**Comment by glenn-jocher** [comment]
@yg-dickson The error suggests there's an issue with how the dataset is being passed to the RKNN toolkit for INT8 calibration. Let's try a few approaches:

1. Make sure you have the latest version of ultralytics:
   ```
   pip install -U ultralytics
   ```

2. Try providing an absolute path to your dataset:
   ```
   yolo export model=best.pt format=rknn int8=True name=rk3588 data=/full/path/to/coco8.yaml
   ```

3. If you're working with a custom model, ensure your dataset.yaml file matches the data structure used during training.

It also appears there might be an issue with how the dataset is being passed to the RKNN toolkit internally. You could try creating an issue in the Ultralytics repository with your complete error log and environment details to help the team investigate further.

**Comment by yg-dickson** [comment]
(base) ubuntu@yingeo-ai:~/Train/ultralytics/runs/detect/e2_yolov8n_320_4.12/weights$ yolo export model=best.pt format=rknn int8=True name=rk3588 data=/home/ubuntu/Train/ultralytics/ultralytics/cfg/dataset 
s/coco8.yaml
Ultralytics 8.3.127 🚀 Python-3.12.8 torch-2.6.0+cpu CPU (-)
Model summary (fused): 72 layers, 3,005,843 parameters, 0 gradients, 8.1 GFLOPs

PyTorch: starting from 'best.pt' with input shape (1, 3, 320, 320) BCHW and output shape(s) (1, 5, 2100) (5.9 MB)

RKNN: starting export with rknn-toolkit2...

ONNX: starting export with onnx 1.16.1 opset 19...
ONNX: slimming with onnxslim 0.1.48...
ONNX: export success ✅ 0.9s, saved as 'best.onnx' (11.6 MB)
I rknn-toolkit2 version: 2.3.2
I Loading : 100%|██████████████████████████████████████████████| 144/144 [00:00<00:00, 70222.04it/s]
E build: If set do_quantization = True or auto_hybrid = True, dataset file need be provided. But it`s value is None!
E export_rknn: RKNN model does not exist, please load & build model first!
RKNN: export success ✅ 1.2s, saved as 'best_rknn_model' (7.1 MB)

**Comment by thinhlx1993** [comment]
Hi @glenn-jocher , i faced the same issue here this is my export log, can you pleas help me check what is wrong here? the export model size is 0Mb looks like it is incorrect.

yolo export model=best-yolov11s.pt format=rknn int8=True name=rk3588 data=/media/data/ssd1/Mixue/YOLODataset/dataset.yaml
Ultralytics 8.3.130 🚀 Python-3.12.4 torch-2.4.0+cu121 CPU (Intel Xeon E5-2698 v3 2.30GHz)
YOLO11s summary (fused): 100 layers, 9,414,348 parameters, 0 gradients, 21.3 GFLOPs

PyTorch: starting from 'best-yolov11s.pt' with input shape (1, 3, 640, 640) BCHW and output shape(s) (1, 8, 8400) (18.3 MB)

RKNN: starting export with rknn-toolkit2...

ONNX: starting export with onnx 1.17.0 opset 19...
ONNX: slimming with onnxslim 0.1.50...
ONNX: export success ✅ 2.2s, saved as 'best-yolov11s.onnx' (36.2 MB)
I rknn-toolkit2 version: 2.3.2
I Loading : 100%|██████████████████████████████████████████████| 196/196 [00:00<00:00, 17336.59it/s]
E build: If set do_quantization = True or auto_hybrid = True, dataset file need be provided. But it`s value is None!
E export_rknn: RKNN model does not exist, please load & build model first!
RKNN: export success ✅ 2.8s, saved as 'best-yolov11s_rknn_model' (0.0 MB)

Export complete (3.5s)
Results saved to /media/data/ssd1/Mixue/YOLODataset
Predict:         yolo predict task=detect model=best-yolov11s_rknn_model imgsz=640 int8 
Validate:        yolo val task=detect model=best-yolov11s_rknn_model imgsz=640 data=dataset.yaml int8 
Visualize:       https://netron.app
💡 Learn more at https://docs.ultralytics.com/modes/export

this is my dataset.yaml

path: /media/data/ssd1/Mixue/YOLODataset
train: train.txt
val: val.txt
test: test.txt

names:
    0: cashbox-open
    1: cash
    2: bill
    3: person

**Comment by LufyCZ** [comment]
This clearly wasn't tested, the build function needs a dataset parameter which is missing.

**Comment by Laughing-q** [comment]
@thinhlx1993 @LufyCZ Thank you guys! We are having another attempt to fix it by adding the dataset argument, and the team is testing it. https://github.com/ultralytics/ultralytics/pull/20508

**Comment by glenn-jocher** [comment]
Closing the loop here: RKNN INT8 export is now supported in Ultralytics starting with `ultralytics>=8.4.59`.

Implementation PR: https://github.com/ultralytics/ultralytics/pull/24656
Platform export UI: https://platform.ultralytics.com/ultralytics/yolo26/yolo26n?tab=export

Python usage now follows the same INT8 calibration-data flow as other export formats, for example:

```python
from ultralytics import YOLO

YOLO("yolo26n.pt").export(format="rknn", name="rk3588", int8=True, data="coco8.yaml")
```

Platform also exposes the RKNN FP16/INT8 export controls directly:

Screenshot 2026-06-01 at 17 31 16
